src/tests/modules/core/outlet_tests.ts

```diff
@@ -348,7 +349,32 @@ export default class OutletTests extends ControllerTestCase(OutletController) {
     )
   }
 
-  async "test outlet disconnected callback when the controlled element's outlet attribute is removed"() {
+  async "test outlet connect callback when the controlled element's outlet attribute is emptied"() {
+    const alpha1 = this.findElement("#alpha1")
+    const alpha2 = this.findElement("#alpha2")
+
+    this.assert.equal(this.controller.alphaOutletConnectedCallCountValue, 2)
+    this.assert.equal(this.controller.alphaOutletDisconnectedCallCountValue, 0)
+
+    await this.setAttribute(this.controller.element, `data-${this.identifier}-alpha-outlet`, "")
+
+    this.assert.equal(this.controller.alphaOutletConnectedCallCountValue, 2)
+    this.assert.equal(this.controller.alphaOutletDisconnectedCallCountValue, 2)
+
+    this.assert.ok(alpha1.isConnected, "alpha1 is still present in document")
+    this.assert.ok(alpha2.isConnected, "alpha2 is still present in document")
+
+    this.assert.ok(
+      alpha1.classList.contains("disconnected"),
+      `expected "${alpha1.className}" to contain "disconnected"`
+    )
+    this.assert.ok(
+      alpha2.classList.contains("disconnected"),
+      `expected "${alpha2.className}" to contain "disconnected"`
+    )
+  }
+
+  async "skip test outlet disconnected callback when the controlled element's outlet attribute is removed"() {
```

Comment:
@seanpdoyle this test case introduced in #624 is now failing due to the nature what this pull request changes.

Just removing the attribute on the controller element would mean that the "fallback selector" would come into play and therefore nothing gets disconnected in this case, because the fallback selector still matches the outlet elements.

Would you agree that this test case is not needed anymore? I adapted this test case to what you see on [L52](https://github.com/hotwired/stimulus/pull/647/files#diff-e52aed23e55a78cd208d17fbb4697756f2375b5aa2027dd00f967ed930fc2eb4R352).

Reply:
What if there were an "atomic" operation that removed the attribute and the referenced element together before an `await this.nextFrame`?

Reply:
Good idea, we could adapt that case to test for this case. Even though I'm not sure if that _could_ happen in a real app.

Reply:
Unskipped in https://github.com/hotwired/stimulus/commit/617a749bef67d1589f93cc6dc25675f5d00b3d9b, thank you!